cypress/integration/transforms_spec.js

```diff
@@ -0,0 +1,247 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { PLUGIN_NAME } from "../support/constants";
+import sampleTransform from "../fixtures/sample_transform";
+
+const TRANSFORM_ID = "test_transform_id";
+
+describe("Transforms", () => {
+    beforeEach(() => {
+      cy.deleteAllIndices();
+
+      // Set welcome screen tracking to test_transform_target
+      localStorage.setItem("home:welcome:show", true);
+
+      // Go to sample data page
+      cy.visit(`${Cypress.env("opensearch_dashboards")}/app/home#/tutorial_directory/sampleData`);
+
+      // Click on "Sample data" tab
+      cy.contains("Sample data", { timeout: 20000 }).click({ force: true });
+      // Load sample eCommerce data
+      cy.get(`button[data-test-subj="addSampleDataSetecommerce"]`).click({ force: true });
+
+      // Verify that sample data is add by checking toast notification
+      cy.contains("Sample eCommerce orders installed", { timeout: 60000 });
+
+      // Visit ISM Transforms Dashboard
+      cy.visit(`${Cypress.env("opensearch_dashboards")}/app/${PLUGIN_NAME}#/transforms`);
+
+      // Common text to wait for to confirm page loaded, give up to 60 seconds for initial load
+      cy.contains("Create transform", { timeout: 60000 });
+    });
+
+    describe("can be created", () => {
+      it("successfully", () => {
+        // Confirm we loaded empty state
+        cy.contains(
+          "Transform jobs help you create a materialized view on top of existing data."
+        );
+
+        // Route to create transform page
+        cy.contains("Create transform").click({ force: true });
+
+        // Type in transform ID
+        cy.get(`input[placeholder="my-transformjob1"]`).type(TRANSFORM_ID, { force: true });
+
+        // Get description input box
+        cy.get(`textarea[data-test-subj="description"]`).focus().type("some description");
+
+        // Enter source index
+        cy.get(`div[data-test-subj="sourceIndexCombobox"]`)
+          .find(`input[data-test-subj="comboBoxSearchInput"]`)
+          .focus()
+          .type("opensearch_dashboards_sample_data_ecommerce{enter}");
+
+        // Enter target index
+        cy.get(`div[data-test-subj="targetIndexCombobox"]`)
+          .find(`input[data-test-subj="comboBoxSearchInput"]`)
+          .focus()
+          .type("target_index{enter}");
+
+        // Click the next button
+        cy.get("button").contains("Next").click({ force: true });
+
+        // Confirm that we got to step 2 of creation page
+        cy.contains("Select fields to transform");
+
+        // Setup Group and aggregation
+        cy.wait(2000);
```

Comment:
Why do we need to wait 2 seconds?

Reply:
Page is dependent on the response from a sample data call on this page which takes a few seconds to load. Adding a delay here prevented a flaky test.

Reply:
I thought the cypress commands already had built in retries/timeouts though?
https://docs.cypress.io/guides/core-concepts/retry-ability#Commands-vs-assertions